app/scripts/lib/rpc-method-middleware/handlers/wallet-createSession/handler.ts

```diff
@@ -0,0 +1,324 @@
+      return Boolean(validScopedProperties?.[scopeString]?.eip3085);
+    };
+
+    const {
+      supportedScopes: supportedRequiredScopes,
+      supportableScopes: supportableRequiredScopes,
+      unsupportableScopes: unsupportableRequiredScopes,
+    } = bucketScopes(supportedRequiredScopesObjects, {
+      isChainIdSupported: existsNetworkClientForChainId,
+      isChainIdSupportable: existsEip3085ForChainId,
+    });
+
+    const {
+      supportedScopes: supportedOptionalScopes,
+      supportableScopes: supportableOptionalScopes,
+      unsupportableScopes: unsupportableOptionalScopes,
+    } = bucketScopes(supportedOptionalScopesObjects, {
+      isChainIdSupported: existsNetworkClientForChainId,
+      isChainIdSupportable: existsEip3085ForChainId,
+    });
+
+    // TODO: placeholder for future CAIP-25 permission confirmation call
+    JSON.stringify({
+      supportedRequiredScopes,
+      supportableRequiredScopes,
+      unsupportableRequiredScopes,
+      supportedOptionalScopes,
+      supportableOptionalScopes,
+      unsupportableOptionalScopes,
+    });
+
+    // Fetch EVM accounts from native wallet keyring
+    // These addresses are lowercased already
+    const existingEvmAddresses = hooks
+      .listAccounts()
+      .map((account) => account.address);
+    const supportedEthAccounts = getEthAccounts({
+      requiredScopes: supportedRequiredScopes,
+      optionalScopes: supportedOptionalScopes,
+    })
+      .map((address) => address.toLowerCase())
+      .filter((address) => existingEvmAddresses.includes(address));
+    const supportedEthChainIds = getPermittedEthChainIds({
+      requiredScopes: supportedRequiredScopes,
+      optionalScopes: supportedOptionalScopes,
+    });
+
+    const legacyApproval = await hooks.requestPermissionApprovalForOrigin({
+      [PermissionNames.eth_accounts]: {
+        caveats: [
+          {
+            type: CaveatTypes.restrictReturnedAccounts,
+            value: supportedEthAccounts,
+          },
+        ],
+      },
+      [PermissionNames.permittedChains]: {
+        caveats: [
+          {
+            type: CaveatTypes.restrictNetworkSwitching,
+            value: supportedEthChainIds,
+          },
+        ],
+      },
+    });
+
+    let caip25CaveatValue = {
+      requiredScopes: getInternalScopesObject(supportedRequiredScopes),
+      optionalScopes: getInternalScopesObject(supportedOptionalScopes),
+      isMultichainOrigin: true,
+      // NOTE: We aren't persisting sessionProperties from the CAIP-25
+      // request because we don't do anything with it yet.
+    };
+
+    caip25CaveatValue = setPermittedEthChainIds(
+      caip25CaveatValue,
+      legacyApproval.approvedChainIds,
+    );
+    caip25CaveatValue = setEthAccounts(
+      caip25CaveatValue,
+      legacyApproval.approvedAccounts,
+    );
+
+    const sessionScopes = getSessionScopes(caip25CaveatValue);
+
+    await Promise.all(
```

Comment:
IIRC we don't actually support these properties yet, so it seems worth it to simplify the PR by removing it

Reply:
removed here https://github.com/MetaMask/metamask-extension/pull/27782/commits/c7f9adc787017225a9f797c9fcda19bbbc1be726